parmed/openmm/parameters.py

```diff
@@ -214,6 +222,19 @@ def _find_unused_types(self, skip_residues):
                     keep_types.add(atom.type)
         return {typ for typ in self.atom_types if typ not in keep_types}
 
+    def _find_identical_residues(self):
+        # this is meant for monoatomic ions ONLY
+        skip_ident_residues = set()
+        monoatomic_types = set()
+        for name, residue in iteritems(self.residues):
+            if len(residue.atoms) == 1:
+                typ = residue.atoms[0].type
```

Comment:
Fantastic point! You reminded me I need to code this for OpenMM - the problem with by-element recognition is what about Fe2+ and Fe3+? I'm perhaps gonna change OpenMM to accept charges for the template matching to resolve that there. But that's also why we have to go either by type here or by (element, charge) - I'd just go with type for simplicity. If you had two monoatomics with the same element, same charge but different types then you're doomed anyway.